omgeo/services/__init__.py

```diff
@@ -658,9 +658,11 @@ def _geocode(self, pq):
                 query = dict(query, searchExtent=pq.viewbox.to_esri_wgs_json())
         else: # single-line
             method = 'find'
+            magic_key = pq.key if hasattr(pq, 'key') else ''
```

Comment:
@CloudNiner, 
Sounds good. I'll do that. 

I did do some testing and my findings are as follows:
The response from the API is identical regardless of the text if ONLY text and magicKey are supplied.

```http://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/find/?text=Nowhere IL&f=json&maxLocations=20&magicKey=GST7YMc0AM9UOsE9HhFtGTyVGST7YMc0AM9UOsE9DbTVHgA9HhB0Zcp0OhNtGMytaikZQswaQDoIM5ktIS9GOghnYnwZGPTq7iAm7Py_CR4PIic2BR4caocpOh9bZgKZQoc3YSyaagDIZhkZZbt0CchC1hp0CbtnZhc0```

is the same as

```http://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/find/?text=the white house&f=json&maxLocations=20&magicKey=GST7YMc0AM9UOsE9HhFtGTyVGST7YMc0AM9UOsE9DbTVHgA9HhB0Zcp0OhNtGMytaikZQswaQDoIM5ktIS9GOghnYnwZGPTq7iAm7Py_CR4PIic2BR4caocpOh9bZgKZQoc3YSyaagDIZhkZZbt0CchC1hp0CbtnZhc0```

but if a bbox is used and the location of the point matching the magicKey is outside the bbox, then the text used and the key is ignored.